src/suspense.js

```diff
@@ -0,0 +1,68 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+// TODO: react warns in dev mode about defaultProps and propTypes not being supported on lazy
+// loaded components
+
+export const sym = Symbol.for('Suspense');
+
+export class Suspense extends Component {
+	constructor(props) {
+		// TODO: should we add propTypes in DEV mode?
+		super(props);
+
+		// mark this component as a handler of suspension (thrown Promises)
+		this[sym] = sym;
+
+		this.state = {
+			l: false
+		};
+	}
+
+	componentDidCatch(e) {
+		if (e && typeof e.then === 'function') {
```

Comment:
Idea: Maybe we can get away with just checking for the presence of a `then` property.

Reply:
Done